src/tests/unit/tests/DetailsView/components/settings-panel/settings-panel.test.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { shallow } from 'enzyme';
+import * as React from 'react';
+import { UserConfigMessageCreator } from '../../../../../../common/message-creators/user-config-message-creator';
+import { BugServicePropertiesMap, UserConfigurationStoreData } from '../../../../../../common/types/store-data/user-configuration-store';
+import { DetailsViewActionMessageCreator } from '../../../../../../DetailsView/actions/details-view-action-message-creator';
+import { SettingsPanel, SettingsPanelProps } from '../../../../../../DetailsView/components/settings-panel/settings-panel';
+
+describe('SettingsPanelTest', () => {
+    let userConfigStoreData: UserConfigurationStoreData;
+
+    type RenderTestCase = {
+        isPanelOpen: boolean;
+        enableTelemetry: boolean;
+        enableHighContrast: boolean;
+        bugService: string;
+        bugServicePropertiesMap: BugServicePropertiesMap;
+    };
+
+    test.each([
+        {
```

Comment:
Any way we can make these tests a bit more readable?

Reply:
how so? The properties of the class are pretty much what you need to pass as properties down the specific settings components.

Reply:
The test case type is gone now. Only moving part now is `isPanelOpen`